.github/workflows/deploy.yml

```diff
@@ -0,0 +1,92 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Deploy to GKE
+
+env:
+  CONFIG_FILE: deploy.toml
+  CLUSTER_NAME: kube
+  CLUSTER_REGION: us-central1
+  REPO_NAME: github-deploy
+
+on:
+  push:
+    branches:
+    - 'main'
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Check out the repository
+      uses: actions/checkout@v3
+
+    - name: Get Google Cloud Credentials
+      uses: google-github-actions/auth@v1
+      with:
+        credentials_json: '${{ secrets.GCP_CREDENTIALS }}'
+        export_environment_variables: true
+
+    - name: Set up Cloud SDK
+      uses: google-github-actions/setup-gcloud@v1
+
+    - name: Give docker access to the GCP registry
+      run: gcloud auth configure-docker ${{ env.CLUSTER_REGION }}-docker.pkg.dev
+
+    - name: Get GKE cluster credentials
+      uses: google-github-actions/get-gke-credentials@v1
+      with:
+        cluster_name: ${{ env.CLUSTER_NAME }}
+        location: ${{ env.CLUSTER_REGION }}
+
+    - name: Install GKE cloud auth plugin
+      run: gcloud components install gke-gcloud-auth-plugin
+
+    - name: Set up Go
+      uses: actions/setup-go@v3
+      with:
+        go-version: '1.20'
+
+    - name: Retrieve weaver version
+      run: |
+        cd 09;
+        echo "WEAVER_VERSION=$(go list -m -f '{{.Version}}' github.com/ServiceWeaver/weaver | sed -E 's/(.*)\..*/\1/')" >> $GITHUB_ENV
+
+    - name: Install weaver-kube
+      run: go install github.com/ServiceWeaver/weaver-kube/cmd/weaver-kube@${{ env.WEAVER_VERSION }}
+
+    - name: Build the emojis app
+      run: cd 09; go build .
+
+    - name: Generate the app config file
+      run: |
+        CONFIG=$(cat << EOF
+        [serviceweaver]
+        binary = "./09/emojis"
+
+        [kube]
+        listeners.emojis = {public = true}
+        image = "${{ env.CLUSTER_REGION }}-docker.pkg.dev/${{ env.GCP_PROJECT }}/${{ env.REPO_NAME }}/emojis"
+        EOF
+        )
+        echo "$CONFIG" > ${{ env.CONFIG_FILE }}
+        cat ${{ env.CONFIG_FILE }}
+
+    - name: Build the YAML files
+      run: weaver-kube deploy ${{ env.CONFIG_FILE }}
+
+    - name: Deploy the YAML files
+      run: |
+        FILE=$(ls *.yaml)
```

Comment:
Is this guaranteed to be a single yaml file?

Reply:
Yes, because we clone everything and run once.